app/api/permissions/specs/routes.spec.ts

```diff
@@ -0,0 +1,134 @@
+import request from 'supertest';
+import { Application, NextFunction, Request, Response } from 'express';
+import { setUpApp } from 'api/utils/testingRoutes';
+import { permissionRoutes } from 'api/permissions/routes';
+import { entitiesPermissions } from 'api/permissions/entitiesPermissions';
+import { collaborators } from 'api/permissions/collaborators';
+
+jest.mock(
+  '../../utils/languageMiddleware.ts',
+  () => (_req: Request, _res: Response, next: NextFunction) => {
+    next();
+  }
+);
+
+describe('permissions routes', () => {
+  let user: { username: string; role: string } | undefined;
+
+  function getUser() {
+    return user;
+  }
+
+  const app: Application = setUpApp(
+    permissionRoutes,
+    (req: Request, _res: Response, next: NextFunction) => {
+      (req as any).user = getUser();
+      next();
+    }
+  );
+
+  describe('entities', () => {
+    describe('POST', () => {
+      beforeEach(() => {
+        spyOn(entitiesPermissions, 'setEntitiesPermissions').and.returnValue(Promise.resolve([]));
+      });
+      it('should save the permissions ', async () => {
+        user = { username: 'user 1', role: 'admin' };
+        const permissionsData = {
+          ids: ['shared1'],
+          permissions: [{ _id: 'user1', type: 'user', level: 'read' }],
+        };
+        const response = await request(app)
+          .post('/api/entities/permissions')
+          .set('X-Requested-With', 'XMLHttpRequest')
+          .send(permissionsData);
+        expect(response.status).toBe(200);
+        expect(entitiesPermissions.setEntitiesPermissions).toHaveBeenCalled();
```

Comment:
i will prefer that the test is an actual integration test, but this is ok too as long as the route has only this one call. but pls in this case use ```.toHaveBeenCalledWith()``` with the expected params

Reply:
done